File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/repository/LockRepository.java
package com.lattis.ellipse.domain.repository;

import com.lattis.ellipse.data.network.model.response.lock.SignedMessagePublicKeyResponse;
import com.lattis.ellipse.domain.model.Lock;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 4/27/17.
 */

public interface LockRepository {

    Observable<SignedMessagePublicKeyResponse> getSignedMessagePublicKey(int bike_id, String macId);
    Observable<Lock> createOrUpdateLock(Lock lock);
    Observable<Lock> getLock();
    Observable<Boolean> deleteLock();
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/repository/LockRepository.kt
package com.lattis.domain.repository

import com.lattis.domain.models.SignedMessageAndPublicKey
import com.lattis.domain.models.Lock
import io.reactivex.rxjava3.core.Observable

interface LockRepository {
    fun getSignedMessagePublicKey(
        bike_id: Int,
        macId: String
    ): Observable<SignedMessageAndPublicKey>

    fun createOrUpdateLock(lock: Lock): Observable<Lock>
    fun getLock(): Observable<Lock>
    fun deleteLock(): Observable<Boolean>

    fun getTapkeyAccess(
        fleet_id: Int,
        mac_id: String
    ): Observable<SignedMessageAndPublicKey>
}